hack/check-state.sh

```diff
@@ -162,13 +162,16 @@ EOF
         if [ "$OPERATION_PROGRESSING" == 'False' ]; then
             set +x
             if [ "$APPLICATION_AVAILABLE" == 'False' ]; then 
-                echo "Error: Operator $kind is not is not available. Detailed status: $APPLICATION_AVAILABLE_DATA"
+                echo "Error: Operator $kind is not available. Detailed status: $APPLICATION_AVAILABLE_DATA"
             fi
             if [ "$APPLICATION_DEGRADED" == 'True' ]; then 
                 echo "Error: Operator $kind is degraded. Detailed status: $APPLICATION_DEGRADED_DATA"
             fi
-            EXIT_STATUS=1
-            return
+            # TODO: revert this once the upgrade gating mechanism is properly handled on HCO side
```

Comment:
"upgrade gating" = aggregating?

Reply:
correctly aggregating conditions is just part of it but we have also to detect that we are upgrading and so set progressing according to that.